File: tests/patterns/python/aws_s3_to_gcs/after.py
from google.cloud import storage

storage_client = storage.Client()

def upload_document(bucket_name, key, content):
    bucket = storage_client.bucket(bucket_name)
    blob = bucket.blob(key)
    blob.upload_from_string(content)

def download_document(bucket_name, key):
    bucket = storage_client.bucket(bucket_name)
    blob = bucket.blob(key)
    return blob.download_as_bytes()

def list_documents(bucket_name, prefix):
    bucket = storage_client.bucket(bucket_name)
    blobs = bucket.list_blobs(prefix=prefix)
    return [blob.name for blob in blobs]

File: tests/patterns/typescript/aws_s3_to_gcs/after.ts
import { Storage } from "@google-cloud/storage";

const storage = new Storage();

export async function uploadDocument(
  bucketName: string,
  key: string,
  content: Buffer
): Promise<void> {
  const bucket = storage.bucket(bucketName);
  const file = bucket.file(key);
  await file.save(content);
}

export async function downloadDocument(
  bucketName: string,
  key: string
): Promise<Buffer> {
  const bucket = storage.bucket(bucketName);
  const file = bucket.file(key);
  const [contents] = await file.download();
  return contents;
}

export async function listDocuments(
  bucketName: string,
  prefix: string
): Promise<string[]> {
  const bucket = storage.bucket(bucketName);
  const [files] = await bucket.getFiles({ prefix });
  return files.map((file) => file.name);
}
